# Project 2
## Joanie Gannon

#### Data from: https://www.kaggle.com/sudalairajkumar/novel-corona-virus-2019-dataset

In [7]:
#Imports
import pandas as pd

---
### Setup Data
---

In [41]:
#Read CSV
df = pd.read_csv("data/novel-corona-virus-2019-dataset/COVID19_open_line_list.csv")
#Look at the info
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14126 entries, 0 to 14125
Data columns (total 45 columns):
ID                          13173 non-null float64
age                         1349 non-null object
sex                         1264 non-null object
city                        10194 non-null object
province                    12906 non-null object
country                     13148 non-null object
wuhan(0)_not_wuhan(1)       13170 non-null float64
latitude                    13147 non-null float64
longitude                   13147 non-null float64
geo_resolution              13147 non-null object
date_onset_symptoms         746 non-null object
date_admission_hospital     730 non-null object
date_confirmation           13089 non-null object
symptoms                    493 non-null object
lives_in_Wuhan              565 non-null object
travel_history_dates        503 non-null object
travel_history_location     758 non-null object
reported_market_exposure    35 non-null object
addi

---
### Dropping unused data.
---
We'll start with an inital pass by dropping these members simply because they hold little or no informational value for us. This is just based on the type of information, and not based on any hard analysis. These rows do not seem to be useful to the analysis we want to do here, regardless of how clean or dirty the data is. We will be dropping:
- lives_in_Wuhan
- geo_resolution
- Additional Information
- Source
- sequence_available
- notes_for_discussion
- Admin
- data_moderator_initials
- Unnamed end columns and any null columns

In [42]:
#Define data we want to drop
dropped_data = ['geo_resolution','source', 'lives_in_Wuhan','additional_information', 'sequence_available',
                'notes_for_discussion', 'admin_id', 'data_moderator_initials','admin3',
                'admin2','admin1']
#Drop uneeded data
df = df.drop(columns = dropped_data, errors = 'ignore')

#clear fully null columns
df = df.dropna(axis = 1, how = 'all')
#This will clear all the padded null rows at the end of the dataset
df = df.dropna(axis = 0, how = 'all')

df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13174 entries, 0 to 13173
Data columns (total 22 columns):
ID                          13173 non-null float64
age                         1349 non-null object
sex                         1264 non-null object
city                        10194 non-null object
province                    12906 non-null object
country                     13148 non-null object
wuhan(0)_not_wuhan(1)       13170 non-null float64
latitude                    13147 non-null float64
longitude                   13147 non-null float64
date_onset_symptoms         746 non-null object
date_admission_hospital     730 non-null object
date_confirmation           13089 non-null object
symptoms                    493 non-null object
travel_history_dates        503 non-null object
travel_history_location     758 non-null object
reported_market_exposure    35 non-null object
chronic_disease_binary      18 non-null float64
chronic_disease             13 non-null object
outcome

#### Now let's see what the data looks like in these columns

In [43]:
df.describe(include='all')

ID    age   sex          city   province country  \
count   13173.000000   1349  1264         10194      12906   13148   
unique           NaN    116     5           806        150      51   
top              NaN  50-59  male  Wenzhou City  Guangdong   China   
freq             NaN     47   703           448       1228   10446   
mean     6769.250436    NaN   NaN           NaN        NaN     NaN   
std      3920.886020    NaN   NaN           NaN        NaN     NaN   
min         1.000000    NaN   NaN           NaN        NaN     NaN   
25%      3351.000000    NaN   NaN           NaN        NaN     NaN   
50%      6865.000000    NaN   NaN           NaN        NaN     NaN   
75%     10180.000000    NaN   NaN           NaN        NaN     NaN   
max     13479.000000    NaN   NaN           NaN        NaN     NaN   

        wuhan(0)_not_wuhan(1)      latitude     longitude date_onset_symptoms  \
count                 13170.0  13147.000000  13147.000000                 746   
unique                    NaN           NaN           NaN                  64   
top                       NaN           NaN           NaN          22.01.2020   
freq                      NaN           NaN           NaN                  43   
mean                      1.0     31.994650    110.495353                 NaN   
std                       0.0      7.536335     27.985144                 NaN   
min                       1.0    -37.813600   -124.953070                 NaN   
25%                       1.0     27.905380    111.729600                 NaN   
50%                       1.0     31.759410    115.441800                 NaN   
75%                       1.0     35.781340    120.440400                 NaN   
max                       1.0     69.649468    153.400000                 NaN   

        ... symptoms travel_history_dates travel_history_location  \
count   ...      493                  503                     758   
unique  ...      217                  123                     178   
top     ...    fever           21.01.2020                   Wuhan   
freq    ...      131                   48                     380   
mean    ...      NaN                  NaN                     NaN   
std     ...      NaN                  NaN                     NaN   
min     ...      NaN                  NaN                     NaN   
25%     ...      NaN                  NaN                     NaN   
50%     ...      NaN                  NaN                     NaN   
75%     ...      NaN                  NaN                     NaN   
max     ...      NaN                  NaN                     NaN   

       reported_market_exposure chronic_disease_binary  \
count                        35              18.000000   
unique                        3                    NaN   
top                          no                    NaN   
freq                         33                    NaN   
mean                        NaN               0.722222   
std                         NaN               0.460889   
min                         NaN               0.000000   
25%                         NaN               0.250000   
50%                         NaN               1.000000   
75%                         NaN               1.000000   
max                         NaN               1.000000   

                                          chronic_disease  outcome  \
count                                                  13      184   
unique                                                 13       12   
top     thought to have had other pre-existing conditions   stable   
freq                                                    1       78   
mean                                                  NaN      NaN   
std                                                   NaN      NaN   
min                                                   NaN      NaN   
25%                                                   NaN      NaN   
50%                                             

#### Looking at these results, we can see a few things that we should investigate:
1) There are 5 unique values for sex, where we would expect 2 or 3

2) Keeping in mind that we have over 13,000 rows, the following have relatively low non-null counts:
- date_onset_symptoms (746)
- date_admission_hospital (730)
- symptoms (493)
- travel_history_dates (503)
- travel_history_location (758)
- reported_market_exposure (35)
- chronic_disease_binary (18)
- chronic_disease (13)
- outcome (184)
- date_death_or_discharge (93)
- location (1024) 

This is unfortunate considering this is about half of our dataset, but we can definetly get away with dropping some of these data points.


---
### Cleaning the data
---
Let's make some of these data members smaller and create categoricals for some other ones.
Categoricals we will be making are:
- Country
- Sex
- Symptoms

In [51]:
#Count how many cases are in each country
df.groupby('country')['ID'].nunique()

country
Afghanistan                 1
Algeria                     1
Australia                  15
Austria                     2
Bahrain                    20
Belgium                     1
Brazil                      1
Cambodia                    1
Canada                     10
China                   10446
Croatia                     3
Ecuador                     1
Egypt                       1
Estonia                     1
Finland                     2
France                     19
Georgia                     2
Germany                    20
Greece                      1
India                       3
Iran                       28
Iraq                        6
Israel                      4
Italy                     588
Japan                     731
Kuwait                     26
Lebanon                     3
Lithuania                   1
Malaysia                   17
Nepal                       1
Netherlands                 2
Nigeria                     1
North Macedonia             1
No

In [16]:
df.head(20).symptoms

0                          NaN
1                          NaN
2                          NaN
3                          NaN
4                          NaN
5                    pneumonia
6                        fever
7                          NaN
8                        fever
9                          NaN
10                   pneumonia
11                         NaN
12                         NaN
13                       cough
14                       fever
15                         NaN
16                         NaN
17                       fever
18    fatigue, fever, headache
19                         NaN
Name: symptoms, dtype: object